lib/chef/node_map.rb

```diff
@@ -287,8 +287,6 @@ def canonical_matches?(canonical, matcher)
     # "provides" lines with identical filters sort by class name (ascending).
     #
     def compare_matchers(key, new_matcher, matcher)
-      cmp = compare_matcher_properties(new_matcher[:target_mode], matcher[:target_mode])
-      return cmp if cmp != 0
```

Comment:
why did we not need this? I think I added it out of copy/pasta.

Reply:
yeah it was a copypasta error...

so what this does is establish the priority and uniqueness of two different rules.

that created problem where i think this line:

```
provides :service, target_mode: true
```

started behaving more like:

```
provides :service { true }
```

so that instead of being the least-specific, lowest-priority match it went to being the highest-priority and difficult to override.  i might not be remembering the case entirely correctly, but it definitely caused some provider_resolver tests to explode.  didn't show up until i started messing with the service provider (which is going to be where all the weird edge cases are).

now the meaning of the flag is identical to what it meant before, but the target_mode flag just means "also supports target mode".

Reply:
(actually its not uniqueness, because there is still that `matches_target_mode?` check, it is just used for sort order -- otherwise deleting those lines would have no-op'd it completely)